bot/admin/admin.py

```diff
@@ -348,6 +353,102 @@ async def change_discord_presence_clear(self, _ctx: commands.Context):
+
+    # @commands.command(name="purge")
+    # @command_log
+    # async def purge_channel(self, ctx: commands.Context, channel: discord.TextChannel):
+    #     """Command handler for the `purge` command.
+    #
+    #     Removes all messages in a channel.
+    #
+    #     Args:
+    #         ctx (discord.ext.commands.Context): The context from which this command is invoked.
+    #         channel (discord.Textchannel): The channel that is to purge
+    #     """
+    #     embed = _build_purge_confirmation_embed(channel)
+    #     timeout = 15.0
+    #     reaction = await self._send_confirmation_dialog(ctx, embed, timeout)
+    #     if reaction is None:
+    #         return
+    #     if str(reaction[0].emoji) == constants.EMOJI_CONFIRM:
+    #         await _purge_channel(channel)
+
+    @commands.Cog.listener(name='on_message')
+    async def on_message(self, ctx: discord.Message):
+        """Event Handler for new messages.
+
+        Deletes a message if the channel it was posted in is in bot-only mode and the author isn't SAM.
+
+        Args:
+            ctx (discord.Message): The context this method was called in. Must always be a message.
+        """
+        if ctx.author == self.bot.user:
+            return
+        if self._db_connector.is_botonly(ctx.channel):
+            await ctx.delete()
+
+    async def _send_confirmation_dialog(self, ctx: commands.Context, embed: discord.Embed, timeout: float) -> \
+            Optional[Tuple[discord.Reaction, discord.User]]:
+        """Handles a confirmation dialog and returns the user reaction.
+
+        Posts an embed and adds reactions for confirmation and cancellation. The reaction that is clicked will be
+        returned. If no reaction is clicked before a timeout has been reached, `None` is returned instead.
+        Regardless of the return value the embed will be deleted shortly after.
+
+        Args:
+            ctx (commands.Context): The context of invokation. Used to send the message.
+            embed (discord.Embed): The embed that will be posted. It should contain some explanation about what will be
+            confirmed.
+            timeout (float): The timeout until the dialog will be canceled
+
+        Returns:
+            (Optional[Tuple[discord.Reaction, discord.User]]):A tuple consisting of a reaction and the user who has
+            reacted. If the dialog runs in the timeout, None is returned.
+        """
+        embed_msg = await ctx.send(embed=embed, delete_after=timeout)
+        await embed_msg.add_reaction(constants.EMOJI_CONFIRM)
+        await embed_msg.add_reaction(constants.EMOJI_CANCEL)
+
+        def check_reaction(_reaction, user):
+            return user == ctx.author and \
+                   str(_reaction.emoji) in [constants.EMOJI_CANCEL, constants.EMOJI_CONFIRM]
+
+        try:
+            reaction = await self.bot.wait_for('reaction_add', timeout=timeout, check=check_reaction)
+        except asyncio.exceptions.TimeoutError:
```

Comment:
Why not let this be handled by the global error handler? The code for this already exists.

Reply:
we already discussed this, I want this method to be able to return None to handle further reaction on a confirmation dialog.